### PR TITLE
Charlesmchen/rework message metadata view

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk 

screen shot 2017-09-27 at 10 27 14 pm
screen shot 2017-09-27 at 10 26 46 pm
screen shot 2017-09-27 at 10 26 33 pm
screen shot 2017-09-27 at 10 26 17 pm
screen shot 2017-09-27 at 10 26 12 pm

### CODE REVIEW

File: Signal/src/ViewControllers/MessageMetadataViewController.swift [hunk 1]
```diff
@@ -147,47 +160,108 @@ class MessageMetadataViewController: OWSViewController {
...
+        if let lastRow = lastRow {
+            lastRow.autoPinEdge(toSuperviewEdge:.bottom, withInset:20)
+        }
+
+        if let mediaMessageView = mediaMessageView {
+            mediaMessageView.autoPinToSquareAspectRatio()
+        }
+    }
+
+    private func addContentRows() -> [UIView] {
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/MessageMetadataViewController.swift]: nit: `addContentRows` reads like a mutator. Should this be just `contentRows()`?

File: Signal/src/ViewControllers/MessageMetadataViewController.swift [hunk 2]
```diff
@@ -305,50 +370,83 @@ class MessageMetadataViewController: OWSViewController {
...
+            let fileSize = dataSource.dataLength()
+            rows.append(valueRow(name: NSLocalizedString("MESSAGE_METADATA_VIEW_ATTACHMENT_FILE_SIZE",
+                                                         comment: "Label for file size of attachments in the 'message metadata' view."),
+                                 value:ViewControllerUtils.formatFileSize(UInt(fileSize))))
+        }
+
+        return rows
+    }
+
+    private func recipientStatus(forOutgoingMessage message: TSOutgoingMessage, recipientId: String) -> (MessageRecipientState, String) {
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/MessageMetadataViewController.swift]: idiomatic nit: `recipientStatus(outgoingMessage: recipientId:)`

**Comment by michaelkirk** [comment]
LGTM!

File: Signal/src/ViewControllers/MessageMetadataViewController.swift (hunk 1 above)
Comment (charlesmchen) [reply Signal/src/ViewControllers/MessageMetadataViewController.swift]: Done.

File: Signal/src/ViewControllers/MessageMetadataViewController.swift (hunk 2 above)
Comment (charlesmchen) [reply Signal/src/ViewControllers/MessageMetadataViewController.swift]: Done.

**Comment by charlesmchen** [comment]
> design nit: I think the "Sent: 9:15" should appear above the list of recipients
> I'm not sure if I'm articulating this very well - but maybe I'm over explaining. Does this make sense?

Sorry, I don't follow.  I moved the "sent" row with the other "metadata" rows down at the bottom because it looks a lot more visually appealing IMO with that grouping.

**Comment by charlesmchen** [comment]
> I'm happy to take a crack at that if you want to pass it off.

Yup, that would be great.  It just looked like a PITA and not worth it.

**Comment by charlesmchen** [comment]
@michaelkirk let's keep iterating on this view, very happy to discuss further refinements.

**Pull request merged**